### PR TITLE
modify_YOLO_object_loss

### PR DESCRIPTION
**Pull request by shkim5616**
First of all, Thank you very much. I can understand YOLO model easily because of your youtube and github code.

In the YOLO original paper, they predict confidence score. If the model is trained well, then the model prediction will close to IoU(Intersection over Union). However, in your code the model prediction will close to 1, if object is existed in the grid cell.
For this problem, I think the loss function(object loss) should be modified. 
In your code, it has already calculated IoU between prediction box and ground truth box. So the modification is going to be simple. Just multiply IoU value in target[...,20:21]. Becuase target[...,20:21] is 1, if the object is existed in the grid cell.

### CODE REVIEW

**Comment by oonisim** [comment]
Related part in the paper.

>  Formally we define confidence as Pr(Object) IOU(truth,pred). If no object exists in that cell, the confidence scores should be zero. Otherwise we want the confidence score to equal the intersection over union (IOU) between the predicted box and the ground truth. Each bounding box consists of 5 predictions: x, y, w, h,and confidence. The (x, y) coordinates represent the center of the box relative to the bounds of the grid cell. The width and height are predicted relative to the whole image. Finally the confidence prediction represents the IOU between the predicted box and any ground truth box.

As pointed out, the confidence ground truth Ci should be calculated using IOU during the training.

> we want the confidence score to **equal the intersection over union (IOU) between the predicted box and the ground truth**
